calculation/gmhazard_calc/gmhazard_calc/directivity/directivity.py

```diff
@@ -157,40 +159,63 @@ def compute_fault_directivity(
         constants.EventType.from_rake(rake),
     )
 
-    # Creating the array to store all fdi values
-    fdi_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    phired_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    for index, planes in enumerate(hypo_planes):
-        # Gets the plane index of the hypocentre
-        plane_index = plane_ind[index]
+    if n_procs == 1:
+        fdi, phired = [], []
+        for ix, planes in enumerate(hypo_planes):
+            cur_fdi, (cur_phi_red, cur_predictor_functions, cur_other) = _compute_directivity_effect(
+                lon_lat_depth,
+                planes,
+                plane_ind[ix],
+                sites,
+                nominal_strike,
+                nominal_strike2,
+                mag,
+                rake,
+                periods,
+            )
 
-        fdi, (phi_red, predictor_functions, other) = _compute_directivity_effect(
-            lon_lat_depth,
-            planes,
-            plane_index,
-            sites,
-            nominal_strike,
-            nominal_strike2,
-            mag,
-            rake,
-            periods,
-        )
+            fdi.append(cur_fdi)
+            phired.append(cur_fdi)
+    else:
+        # Compute directivity for all hypocentres
+        with mp.Pool(n_procs) as pool:
+            results = pool.starmap(
+                _compute_directivity_effect,
+                [
+                    (
+                        lon_lat_depth,
+                        planes,
+                        plane_ind[ix],
+                        sites,
+                        nominal_strike,
+                        nominal_strike2,
+                        mag,
+                        rake,
+                        periods,
+                    )
+                    for (ix, cur_plane) in enumerate(hypo_planes)
+                ],
+            )
+
+        # Combine results
+        fdi = [cur_result[0] for cur_result in results]
+        phired = [cur_result[1][0] for cur_result in results]
```

Comment:
Could this be done like this?
`fdi, phired = list(zip(*results))`
Just feels cleaner to me

Reply:
I think it can't be? if the period is `cur_result[1]` then might work instead of `cur_result[1]` 🤔

Reply:
Hmh maybe for `fdi` something like `fdi = list(list(zip(results))[0])` would work, but it
a) wouldn't work for `phired`
b) I don't think this is any cleaner, much harder to read. 
One of the reasons why I list list comprehensions are pretty simple to read (once you are know the syntax) compared to things like `apply` etc.